src/OpenTelemetry/Trace/Span.cs

```diff
@@ -42,6 +42,22 @@ public sealed class Span : ISpan
         private DateTimeOffset endTimestamp;
         private bool hasEnded;
 
+        private Span(
+            string name,
+            SpanContext parentSpanContext,
+            ActivityAndTracestate activityAndTracestate,
+            bool ownsActivity,
+            SpanKind spanKind,
+            DateTimeOffset startTimestamp,
+            Func<IEnumerable<Link>> linksGetter,
+            TracerConfiguration tracerConfiguration,
+            SpanProcessor spanProcessor,
+            Resource libraryResource)
+        : this(name, parentSpanContext, activityAndTracestate, ownsActivity, spanKind, startTimestamp,
+            linksGetter?.Invoke() ?? Array.Empty<Link>(), tracerConfiguration, spanProcessor, libraryResource)
```

Comment:
don't fallback to empty array - null is not bad

Reply:
Will close this with whatever answer given to the point above.